tests/general/base.py

```diff
@@ -0,0 +1,200 @@
+from test_plus import APITestCase
+from django.core.cache import cache
+
+from tests.profile.factories import (
+    ProfileFactory, ProfileHighlightFactory,
+    ProfileKeyMetricsFactory, IndicatorCategoryFactory,
+    IndicatorSubcategoryFactory
+)
+from tests.points.factories import (
+    ProfileCategoryFactory, ThemeFactory, CategoryFactory, LocationFactory
+)
+from tests.datasets.factories import (
+    GeographyFactory, GeographyHierarchyFactory,
+    VersionFactory, DatasetFactory, GroupFactory,
+    DatasetDataFactory, IndicatorFactory
+)
+from tests.boundaries.factories import GeographyBoundaryFactory
+from wazimap_ng.datasets.models import Group, Geography
+from wazimap_ng.datasets.tasks.indicator_data_extraction import (
+    indicator_data_extraction
+)
+
+
+
+class ConsolidatedProfileViewBase(APITestCase):
+
+    def setUp(self):
+        self.version = VersionFactory()
+        self.root = Geography.add_root(code="ZA", level="country")
+        self.geo1 = self.root.add_child(code="WC", level="province")
+        self.geo2 = self.root.add_child(code="EC", level="province")
+        self.dc1 = self.geo1.add_child(code="DC1", level="muni")
+        self.create_boundary(self.root, self.version)
+        self.create_boundary(self.geo1, self.version)
+        self.create_boundary(self.geo2, self.version)
+
+        self.hierarchy = self.create_hierarchy(
+            self.root, self.version
+        )
+        self.profile = ProfileFactory(geography_hierarchy=self.hierarchy)
+        self.category = CategoryFactory(profile=self.profile)
+        self.location = LocationFactory(category=self.category)
+
+    def tearDown(self):
+        cache.clear()
+
+    def create_boundary(self, geography, version):
+        return GeographyBoundaryFactory(
+            geography=geography, version=version
+        )
+
+    def create_hierarchy(self, root, version=None, **kwargs):
+        configuration = kwargs
+        if version:
+            configuration["default_version"] = version.name
+
+        return GeographyHierarchyFactory(
+            root_geography=root,
+            configuration=configuration
+        )
+
+    def create_pc(self, theme, order=0, label="pc label"):
+        return ProfileCategoryFactory(
+            profile=self.profile, category=self.category, theme=theme,
+            label=label, order=order
+        )
+
+    def create_multiple_profile_categories(self, theme, count=2):
+        pcs = []
+        for i in range(0, count):
+            pcs.append(
+                self.create_pc(theme, i, F"pc{i}_{theme.name}")
+            )
+        return pcs
+
+    def create_dataset(self, profile, version):
+        return DatasetFactory(
+            profile=profile, version=version
+        )
```

Comment:
why not just call the constructor?

Reply:
`create_dataset ` is being called by `create_versioned_data` that is creating versioned data for possible a diff profile or version each time it is called.